### DISCUSSION TITLE
Issue with Mageworx Advanced Product Option

### DISCUSSION THREAD

**Discussion by Officesleek** [breezefront/community#285] [3. Integrations]
Hello,

We are currently expereincing issue with Mageworx Advanced Product Option. This issue is with the Option Swatch. When Breeze Turbo feature is enabled, it returns some errors on the console and product options are not able to be selected except you reload the page. But when Turbo feature is disabled, everything works fine. Note that we have applied all the patches and have the Breeze integration module for Mageworx Advanced Product Option installed.

Uncaught TypeError: name is undefined
    findOptionId https://officesleek.com/static/version1718032282/frontend/Local/Officesleek/en_US/Swissup_Breeze/bundles/1/default1.js:2207
    processSwatchLabel https://officesleek.com/static/version1718032282/frontend/Local/Officesleek/en_US/Swissup_Breeze/bundles/1/product2.js:1053
    update https://officesleek.com/static/version1718032282/frontend/Local/Officesleek/en_US/Swissup_Breeze/bundles/1/product2.js:1108
    onOptionChanged https://officesleek.com/static/version1718032282/frontend/Local/Officesleek/en_US/Swissup_Breeze/bundles/1/product.js:2421
    d https://officesleek.com/static/version1718032282/frontend/Local/Officesleek/en_US/Swissup_Breeze/bundles/1/core.js:14
    each https://officesleek.com/static/version1718032282/frontend/Local/Officesleek/en_US/Swissup_Breeze/bundles/1/core.js:495
    Zn https://officesleek.com/static/version1718032282/frontend/Local/Officesleek/en_US/Swissup_Breeze/bundles/1/core.js:13

**Comment by vovayatsyuk** [comment]
Can you share the URL to the product page where the error occurs?

**Reply by Officesleek** [reply]
Hi Vovayatsyuk,
Thanks for your reply. We had already disabled the Mageworx Option Swatch as it was not allowing customers place order. https://officesleek.com/thf-long-sleeve-shirt-3.html

**Reply by vovayatsyuk** [reply]
Can you create test product with an option swatch so I can debug it on your site?

**Reply by Officesleek** [reply]
Ok

**Reply by Officesleek** [reply]
https://officesleek.com/sleek-fashion.html

You can check this link and click on any of the first two products shown in this screenshot. Open any of the products and try selection swatch option.
![Screenshot_103](https://github.com/breezefront/community/assets/168022721/2b5ebef3-e24e-4e75-87c8-884ec25fc4b9)

**Reply by vovayatsyuk** [reply]
Thank you!

**Comment by vovayatsyuk** [comment]
Fixed: https://github.com/breezefront/module-breeze/commit/e8ecce5793defd5c02cc3e660bf246f0f71a431e

**Reply by Officesleek** [reply]
Thank you! Everything works fine after the update.

**Comment by Officesleek** [comment]
Hello Vovayatsyuk,

I would like to find out if this intergartion module also handles "Replace Main Gallery Image" functionality. At the moment this does not work on Evolution theme. 

https://officesleek.com/adidas-f2-retropy-1.html
![Screenshot_105](https://github.com/breezefront/community/assets/168022721/09a957dd-16eb-41fc-9498-fa430f50b089)

**Reply by vovayatsyuk** [reply]
Try updating to the latest versions of Breeze and integration module. I think I fixed this issue too.

**Reply by Officesleek** [reply]
Thanks for your reply. We have updated to the latest versions of Breeze and Mageworx integration module. The Replace Main Gallery Image does not work. Other features work just fine. You can check the test product via the link below.

https://officesleek.com/test-product.html

**Reply by vovayatsyuk** [reply]
Make sure you configured the module correctly. It's not working now because `this.isEnabledOptionReplaceMode()` returns false.

**Reply by Officesleek** [reply]
My bad! I did not enable the image replacer from the option settings. I am really sorry for wasting your time. Thank you!
